celery/events/state.py

```diff
@@ -296,7 +296,7 @@ def __init__(self, uuid=None, cluster_state=None, children=None, **kwargs):
         self.children = WeakSet(
             self.cluster_state.tasks.get(task_id)
             for task_id in children or ()
-            if task_id in self.cluster_state.tasks
+            if self.cluster_state is not None and task_id in self.cluster_state.tasks
```

Comment:
It looks like it's a better idea to extract the additional condition outside of the comprehension.